graylog2-server/src/main/java/org/graylog2/inputs/transports/UdpTransport.java

```diff
@@ -16,66 +16,135 @@
-        this.workerExecutor = executorService("worker", "udp-transport-worker-%d", localRegistry);
+        super(configuration, eventLoopGroupFactory, throughputCounter, localRegistry);
+        this.nettyTransportConfiguration = nettyTransportConfiguration;
+        this.channels = new DefaultChannelGroup(GlobalEventExecutor.INSTANCE);
     }
 
-    private static Executor executorService(final String executorName, final String threadNameFormat, final LocalMetricRegistry localRegistry) {
-        final ThreadFactory threadFactory = new ThreadFactoryBuilder().setNameFormat(threadNameFormat).build();
-        return new InstrumentedExecutorService(
-                Executors.newCachedThreadPool(threadFactory),
-                localRegistry,
-                name(UdpTransport.class, executorName, "executor-service"));
+    @VisibleForTesting
+    Bootstrap getBootstrap(MessageInput input) {
+        LOG.debug("Setting UDP receive buffer size to {} bytes", getRecvBufferSize());
+        final NettyTransportType transportType = nettyTransportConfiguration.getType();
+
+        eventLoopGroup = eventLoopGroupFactory.create(workerThreads, MetricRegistry.name(this.getClass(), "workers"));
+
+        return new Bootstrap()
+                .group(eventLoopGroup)
+                .channelFactory(new DatagramChannelFactory(transportType))
+                .option(ChannelOption.RCVBUF_ALLOCATOR, new FixedRecvByteBufAllocator(getRecvBufferSize()))
+                .option(ChannelOption.SO_RCVBUF, getRecvBufferSize())
+                .option(UnixChannelOption.SO_REUSEPORT, true)
+                .handler(getChannelInitializer(getChannelHandlers(input)))
+                .validate();
+    }
+
+    @Override
+    protected LinkedHashMap<String, Callable<? extends ChannelHandler>> getChannelHandlers(MessageInput input) {
+        final LinkedHashMap<String, Callable<? extends ChannelHandler>> handlers = new LinkedHashMap<>(super.getChannelHandlers(input));
+        handlers.put("traffic-counter", () -> throughputCounter);
+        handlers.put("udp-datagram", () -> DatagramPacketHandler.INSTANCE);
+        handlers.putAll(getChildChannelHandlers(input));
+
+        return handlers;
     }
 
     @Override
-    public Bootstrap getBootstrap() {
-        final ConnectionlessBootstrap bootstrap = new ConnectionlessBootstrap(new NioDatagramChannelFactory(workerExecutor));
+    public void launch(final MessageInput input) throws MisfireException {
+        try {
+            bootstrap = getBootstrap(input);
 
-        final int recvBufferSize = Ints.saturatedCast(getRecvBufferSize());
-        LOG.debug("Setting receive buffer size to {} bytes", recvBufferSize);
-        bootstrap.setOption("receiveBufferSizePredictorFactory", new FixedReceiveBufferSizePredictorFactory(recvBufferSize));
-        bootstrap.setOption("receiveBufferSize", recvBufferSize);
+            // TODO: Make number of channels configurable separately
```

Comment:
Is this a todo for this PR or a "nice to have"?

Reply:
It's outdated.